modules/backend/behaviors/ListController.php

```diff
@@ -362,6 +364,103 @@ public function index_onDelete()
         return $this->controller->listRefresh($definition);
     }
 
+    /**
+     * Bulk copy records.
+     * @return void
+     * @throws \Winter\Storm\Exception\ApplicationException when the parent definition is missing.
+     */
+    public function index_onCopy()
+    {
+        if (method_exists($this->controller, 'onCopy')) {
+            return call_user_func_array([$this->controller, 'onCopy'], func_get_args());
+        }
+
+        /*
+         * Establish the list definition
+         */
+        $definition = post('definition', $this->primaryDefinition);
+
+        if (!isset($this->listDefinitions[$definition])) {
+            throw new ApplicationException(Lang::get('backend::lang.list.missing_parent_definition', compact('definition')));
+        }
+
+        $listConfig = $this->controller->listGetConfig($definition);
+
+        /*
+         * Validate checked identifiers
+         */
+        $checkedIds = post('checked', []);
+
+        if (!is_array($checkedIds) || !count($checkedIds)) {
+            Flash::error(Lang::get(
+                (!empty($listConfig->noRecordsCopiedMessage))
+                    ? $listConfig->noRecordsCopiedMessage
+                    : 'backend::lang.list.copy_selected_empty'
+            ));
+            return $this->controller->listRefresh();
+        }
+
+        /*
+         * Create the model
+         */
+        $class = $listConfig->modelClass;
+        $model = new $class;
+        $model = $this->controller->listExtendModel($model, $definition);
+
+        /*
+         * Create the query
+         */
+        $query = $model->newQuery();
+        $this->controller->listExtendQueryBefore($query, $definition);
+
+        $query->whereIn($model->getKeyName(), $checkedIds);
+        $this->controller->listExtendQuery($query, $definition);
+
+        /*
+         * Retrieve protected columns from list column configuration
+         */
+        $columnConfig = $this->makeConfig($listConfig->list);
+
+        $protectedColumns = [];
+
+        foreach ($columnConfig as $column => $config) {
+            if (array_key_exists('protected', $config) && $config['protected'] === true) {
+                $protectedColumns[] = $column;
+            }
+        }
```

Comment:
I would prefer if the configuration was available under `$listConfig->list` and supported either an explicit allow list or an explicit ignore list (example below):

`config_list.yaml`
```yaml
replication:
    enabled: true
    # one of the following, if both present then the most restrictive combination should be used
    allowed: [name, type, status]
    ignored: [id, slug, password]
```

The allowed functionality can probably be implemented by doing `$protected = array_diff($allowed, array_keys($record->getAttributes())`

And if replication is enabled and neither is provided then we should default to looking at the model's fillable / guarded definitions (allowing explicit configuration in the config_list.yaml to overrule those values if desired).

@bennothommo any thoughts?

Reply:
@bennothommo further idea, should model replication be controlled solely through properties on the model itself, so we can then more easily reuse this logic for a "Duplicate" button in the form context for duplicating records one by one?

Reply:
I think we try and stick close to what Laravel has done, for feature parity. Laravel's `replicate()` method accepts fields as an "exclude" list, ie. exclude these fields from being replicated - I guess the assumption is that you don't want to have to have yet another array of fields you have to add new fields to. I also think that tying it to the `guarded` / `fillable` arrays might dilute what `replicate` is supposed to do - it's supposed to replicate an entire model after all.

Reply:
To answer your question about keeping the control within the model - I think that would be OK, but I would: a) allow the list controller to override this if a list of fields is provided, and b) stick with Laravel and make it an exclude list.

Reply:
We should also have an event fired on the model on replication, if there isn't one already.

Reply:
What concern do you have about also supporting allow lists rather than just ignore lists? Laravel's replicate() as implemented is a low level feature where developers have to explicitly call it and know what they're doing, this feature in Winter is meant for inclusion in backend lists (potentially all of them by default) and use by end users

Reply:
@bennothommo also what is your proposed implementation for the event? I'm fine leaving it out for now if we don't have a specific design for it yet.

Reply:
I don't see a purpose for an allow list. This is a copy function - in most cases, it will copy everything. If we add an allow list, that's an extra bit of functionality on top of Laravel that we need to track in the future for breakages on major updates.

Reply:
In conclusion what I am supposed to do in this pull request ?

Reply:
@marvindurot if you're still interested in having this merged then just ping me and I'll reopen. If you could submit a PR to the docs that should be all that's needed at this point.